src/components/PopupWithForm.js

```diff
@@ -14,13 +15,19 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  open() {
+    this._submitButton.textContent = "Save";
+    super.open();
+  }
+
   close() {
     this._popupForm.removeEventListener("submit", this._handleSubmitButton);
     this._popupForm.reset();
     super.close();
   }
 
   _handleSubmitButton = (e) => {
+    this._submitButton.textContent = "Saving...";
     e.preventDefault();
     this._handleFormSubmit(this._getInputValues());
     this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in index.js, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.